distributed/client.py

```diff
@@ -1773,18 +1771,7 @@ def close(self, timeout=no_default):
                 coro = wait_for(coro, timeout)
             return coro
 
-        if self._start_arg is None:
-            with suppress(AttributeError):
-                f = self.cluster.close()
-                if asyncio.iscoroutine(f):
-
-                    async def _():
-                        await f
-
-                    self.sync(_)
-
```

Comment:
This whole block is redundant with the same inside `Client._close`.
This changes the `close()` and `__exit__()` behaviour of a synchronous Client that started its own LocalCluster (e.g. `client = Client()`), so that the client first closes itself, thus releasing all tasks on the scheduler, and only afterwards it closes the cluster instead of the other way around.

Without this, as the workers close, the scheduler would observe the sudden loss of all tasks the client had in `who_wants`.
See test: `test_quiet_client_close`